src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/ModuleFreeCam.kt

```diff
@@ -0,0 +1,108 @@
+    private val resetMotion by boolean("ResetMotion", true)
+    private val spoofMovement by boolean("SpoofMovement", true)
+
+    private var fakePlayer: OtherClientPlayerEntity? = null
+    private var x = 0.0
+    private var y = 0.0
+    private var z = 0.0
+
+    override fun enable() {
+        if (mc.player == null) {
+            return
+        } else {
+            if (resetMotion) {
+                player.setVelocity(0.0, 0.0, 0.0)
+            } else {
+                x = player.velocity.x
+                y = player.velocity.y
+                z = player.velocity.z
+            }
+            val faker = OtherClientPlayerEntity(world, player.gameProfile)
+
+            faker.headYaw = player.headYaw
+            faker.copyPositionAndRotation(player)
+            world.addEntity(faker.entityId, faker)
+            fakePlayer = faker
+
+            if (!collision) {
+                player.noClip = true
+            }
+        }
+    }
+
+    override fun disable() {
+        if (mc.player == null || mc.world == null) {
+            return
+        } else {
+            player.updatePositionAndAngles(fakePlayer!!.x, fakePlayer!!.y, fakePlayer!!.z, player.yaw, player.pitch)
+            world.removeEntity(fakePlayer!!.entityId)
+            fakePlayer = null
+            player.setVelocity(x, y, z)
+        }
+    }
+
+    val repeatable = repeatable {
+        if (mc.world == null) {
+            enabled = false
+        } else {
+            // Just to make sure it stays enabled
+            if (!collision) {
+                player.noClip = true
+                player.fallDistance = 0f
+                player.isOnGround = false
+            }
+
+            if (fly) {
+                val speed = speed.toDouble()
+                if (player.moving) {
+                    player.strafe(speed = speed)
+                }
+
+                player.velocity.y = when {
+                    mc.options.keyJump.isPressed -> speed
+                    mc.options.keySneak.isPressed -> -speed
+                    else -> 0.0
+                }
+            }
+        }
+    }
+
+    val packetHandler = handler<PacketEvent> { event ->
+        when (val packet = event.packet) {
+            // For better FreeCam detecting AntiCheats, we need to prove to them that the player's moving
+            is PlayerMoveC2SPacket.PositionOnly, is PlayerMoveC2SPacket.LookOnly, is PlayerMoveC2SPacket.Both -> {
+                if (spoofMovement) {
+                    network.sendPacket(PlayerMoveC2SPacket(fakePlayer!!.isOnGround))
+                    event.cancelEvent()
+                }
+            }
+            is PlayerMoveC2SPacket, is PlayerActionC2SPacket -> event.cancelEvent()
+            // In case of a teleport
+            is PlayerPositionLookS2CPacket -> {
+                fakePlayer!!.setPos(packet.x, packet.y, packet.z)
+                // Reset the motion
+                player.setVelocity(0.0, 0.0, 0.0)
```

Comment:
huh why reset player motion
i think u need to reset recorded motion
~~~
x=0.0
y=0.0
z=0.0
~~~

Reply:
Why reset a variable?